rust/src/lib.rs

```diff
@@ -101,6 +121,14 @@ pub extern "C" fn zcashlc_last_error_length() -> i32 {
 }
 
 /// Copies the last error message into the provided allocated buffer.
+///
+/// # Safety
+///
+/// - `buf` must be non-null and valid for reads for `length` bytes, and it must have an alignment
+///   of `1`.
+/// - The memory referenced by `buf` must not be mutated for the duration of the function call.
+/// - The total size `length` must be no larger than `isize::MAX`. See the safety documentation of
+///   pointer::offset.
```

Comment:
This should be a separate refactor, but I'm really not a fan of stateful error handling using a thread-local variable to hold the error state (as implemented by the [ffi_helpers::error_handling](https://docs.rs/ffi_helpers/latest/ffi_helpers/error_handling/index.html) module). Once we have https://github.com/zcash/zcash/issues/4816 (Define a "result" type for use in C++ error handling) we should replace this error handling with that.

Reply:
Filed as #37.

Reply:
I'm afraid this won't really work, because the other side of this FFI is not C++; it's Swift.